src/components/common/BtnTask/BtnTask.tsx

```diff
@@ -0,0 +1,214 @@
+import { css, Theme } from '@emotion/react';
+import styled from '@emotion/styled';
+import React, { useState } from 'react';
+
+import BtnDate from '../BtnDate/BtnDate';
+import StatusDoneBtn from '../button/statusBtn/StatusDoneBtn';
+import StatusInProgressBtn from '../button/statusBtn/StatusInProgressBtn';
+import StatusStagingBtn from '../button/statusBtn/StatusStagingBtn';
+import StatusTodoBtn from '../button/statusBtn/StatusTodoBtn';
+
+import Icons from '@/assets/svg/index';
+
+interface BtnTaskProps {
+	btnType: number;
+	status?: string;
+	isDescription?: boolean;
+}
+
+interface BorderColorProps {
+	isHovered: boolean;
+	isClicked: boolean;
+	iconHovered: boolean;
+	theme: Theme;
+	btnType: number;
+}
+
+function BtnTask(props: BtnTaskProps) {
+	const { btnType, status = 'Todo', isDescription = false } = props;
+	const [isClicked, setIsClicked] = useState(false);
+	const [isHovered, setIsHovered] = useState(false);
+	const [iconHovered, setIconHovered] = useState(false);
+	const [iconClicked, setIconClicked] = useState(false);
+
+	const handleMouseEnter = () => {
+		setIsHovered(true);
+	};
+
+	const handleMouseLeave = () => {
+		setIsHovered(false);
+	};
+
+	const handleClick = () => {
+		setIsClicked((prev) => !prev);
+	};
+
+	const stopPropagation = (e: React.MouseEvent) => {
+		e.stopPropagation();
+	};
+
+	const handleIconMouseEnter = () => {
+		setIconHovered(true);
+	};
+
+	const handleIconMouseLeave = () => {
+		setIconHovered(false);
+	};
+
+	const handleIconClick = (e: React.MouseEvent) => {
+		if (iconClicked) {
+			setIconClicked(false);
+		} else {
+			stopPropagation(e);
+			setIconClicked(true);
+		}
+	};
+
+	const renderStatusButton = () => {
+		if (btnType === 3) {
+			return <StagingIconHoverIndicator />;
+		}
+
+		if (!iconHovered && !iconClicked) {
+			return <IconHoverIndicator />;
+		}
+
+		if (btnType === 1) {
+			return <StatusStagingBtn />;
+		}
+
+		if (btnType === 2) {
+			if (status === 'Done') {
+				return <StatusDoneBtn />;
+			}
+			if (status === 'InProgress') {
+				return <StatusInProgressBtn />;
+			}
+			return <StatusTodoBtn />;
+		}
+
+		return null;
+	};
+
+	return (
+		<BtnTaskLayout
+			isClicked={isClicked}
+			isHovered={isHovered}
+			iconHovered={iconHovered}
+			btnType={btnType}
+			onClick={handleClick}
+		>
+			<BtnTaskContainer onMouseEnter={handleMouseEnter} onMouseLeave={handleMouseLeave}>
+				<BtnTaskTextWrapper isDescription={isDescription}>
+					{isDescription && <IconFile />}
+					넛쉘 UT 진행하기
+				</BtnTaskTextWrapper>
+				{btnType === 3 ? (
+					<BtnDate date="2024.07.11" time="22:22" size="small" isDelayed />
+				) : (
+					<BtnDate date="2024.07.11" time="22:22" size="small" />
+				)}
```

Comment:
```suggestion
								<BtnDate date="2024.07.11" time="22:22" size="small" isDelayed={btnType === 3} />
```

isDelayed 속성 안에서 btnType 3인지 판단하면 더 간단할 것 같습니당

Reply:
레전드 간단해졌다 코리 보면서 배우는 중.,, 👍

Reply:
진짜 👍 👍  바로 적용했슴다